test/definition_name_test.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:rohd/rohd.dart';
+import 'package:rohd/src/exception/name_exception.dart';
+import 'package:rohd/src/utilities/sanitizer.dart';
+import 'package:test/test.dart';
+
+class DefinitionName {
+  String? name;
+  bool isReserved;
+
+  DefinitionName({required this.name, required this.isReserved});
+
+  static String getValidName() => 'specialName';
+  static String getInvalidName() => '/--**definitionName+';
+}
+
+class ValidDefNameModule extends Module {
+  ValidDefNameModule(Logic a, DefinitionName defName)
+      : super(
+          name: 'specialNameInstance',
+          reserveName: false,
+          definitionName: defName.name,
+          reserveDefinitionName: defName.isReserved,
+        ) {
+    addInput('a', a, width: a.width);
+  }
+}
+
+class RunTimeNameModule extends ValidDefNameModule {
+  String mockRunTimeString;
+
+  RunTimeNameModule(super.a, super.defName, this.mockRunTimeString);
+  @override
+  String get definitionName {
+    const String? mockDefinitionName = null;
+
+    return Sanitizer.sanitizeSV(mockDefinitionName ?? mockRunTimeString);
```

Comment:
What do you test of your changes in `Module` if you override `definitionName`?

Reply:
```
class byte
```


create a reservedName class to test for runtime type

Reply:
Added reserved name class and tested.